src/igz_controls/services/validation.service.js

```diff
@@ -0,0 +1,579 @@
+                    label: $i18next.t('common:BEGIN_NOT_WITH', {lng: lng}) + ': ' + convertToLabel(chars),
+                    pattern: new RegExp('^[^' + convertToPattern(chars) + ']')
+                };
+            },
+            endWith: function (chars) {
+                return {
+                    name: 'end',
+                    label: $i18next.t('common:END_WITH', {lng: lng}) + ': ' + convertToLabel(chars),
+                    pattern: new RegExp('[' + convertToPattern(chars) + ']$')
+                };
+            },
+            beginEndWith: function (chars) {
+                var convertedPattern = convertToPattern(chars);
+
+                return {
+                    name: 'beginEnd',
+                    label: $i18next.t('common:BEGIN_END_WITH', {lng: lng}) + ': ' + convertToLabel(chars),
+                    pattern: new RegExp('^([' + convertedPattern + '].*)?[' + convertedPattern + ']$')
+                };
+            },
+            beginEndNotWith: function (chars) {
+                var convertedPattern = convertToPattern(chars);
+
+                return {
+                    name: 'beginEndNot',
+                    label: $i18next.t('common:BEGIN_END_NOT_WITH', {lng: lng}) + ': ' + convertToLabel(chars),
+                    pattern: new RegExp('^([^' + convertedPattern + '].*)?[^' + convertedPattern + ']$')
+                };
+            },
+            validCharacters: function (chars) {
+                return {
+                    name: 'validCharacters',
+                    label: $i18next.t('common:VALID_CHARACTERS', {lng: lng}) + ': ' + convertToLabel(chars),
+                    pattern: new RegExp('^[' + convertToPattern(chars) + ']+$')
+                };
+            },
+            length: function (options) {
+                var min = Number.isSafeInteger(options.min) ? options.min : 0;
+                var max = Number.isSafeInteger(options.max) ? options.max : '';
+
+                if (min || max) {
+                    var label = $i18next.t('common:LENGTH', {lng: lng}) + ' – ' +
+                        (min ? 'min: ' + options.min + '\xa0\xa0' : '') + (max ? 'max: ' + options.max : '');
+
+                    return {
+                        name: 'length',
+                        label: label,
+                        pattern: new RegExp('^[\\S\\s]{' + min + ',' + max + '}$')
+                    };
+                }
+            },
+            noConsecutiveCharacters: function (chars) {
+                var convertedPattern = chars.split(' ').map(function (charPair) {
+                    var charsPairArray = charPair.split('');
+
+                    return '(?!.*' + '\\' + charsPairArray[0] + '\\' + charsPairArray[1] + ')';
+                }).join('');
+
+                return {
+                    name: 'noConsecutiveCharacters',
+                    label: $i18next.t('common:NO_CONSECUTIVE_CHARACTER', {lng: lng}) + ': (' + convertToLabel(chars) + ')',
```

Comment:
Why we suddenly have `(` and `)` here?
![image](https://user-images.githubusercontent.com/13918850/81794271-338cbc80-9513-11ea-827b-63b5437cffa7.png)